modules/core/shared/src/main/scala/net/protocol/ParseDescribe.scala

```diff
@@ -44,14 +44,18 @@ object ParseDescribe {
                 ).raiseError[F, Unit]
         } yield a
 
-      def parseExchange(stmt: Statement[_], ty: Typer)(span: Span[F]): F[(F[StatementId], StatementId => F[Unit])] = 
+      def parseExchange(stmt: Statement[_], ty: Typer)(span: Span[F]): F[(F[StatementId], StatementId => F[Unit], F[Unit])] = 
         stmt.encoder.oids(ty) match {
 
           case Right(os) if os.length > Short.MaxValue =>
-            TooManyParametersException(stmt).raiseError[F, (F[StatementId], StatementId => F[Unit])]
+            TooManyParametersException(stmt).raiseError[F, (F[StatementId], StatementId => F[Unit], F[Unit])]
 
           case Right(os) =>
-            OptionT(parseCache.value.get(stmt)).map(id => (id.pure, (_:StatementId) => ().pure)).getOrElse {
+
+            val closeEvictedStatements =
```

Comment:
Extracted here b/c this is needed in both the `get` and `insert` cases below (because `get` can cause eviction of `gen1` in case where target entry is in `gen1`).